### PR TITLE
chore(flags): filter unrelated flags out of the match section

### PR DESCRIPTION
**Pull request by dmarticus**
## Problem

I don't want to show flags that don't match people in the "people" section where we show the feature flags for which they're associated – if a flag doesn't match on person properties, I don't think we should show that to our users.  Context: https://posthog.slack.com/archives/C07Q2U4BH4L/p1731539410714099

## Changes

- filters group flags from person properties
- filters person flags from group properties
- add tooltips to each section explaining what they should expect

Loom: https://www.loom.com/share/af638a71d0674572bcf15b44b92c4108

## How did you test this code?

Visual UI tests, didn't think it was worth a cypress test.  Maybe worth a logic test?  Idk.

### CODE REVIEW

File: frontend/src/scenes/persons/RelatedFeatureFlags.tsx [hunk 1]
```diff
@@ -123,7 +124,11 @@ export function RelatedFeatureFlags({ distinctId, groups }: Props): JSX.Element
                 const matchesSet = featureFlag.evaluation.reason === FeatureFlagMatchReason.ConditionMatch
                 return (
                     <div>
-                        {featureFlag.active ? <>{featureFlagMatchMapping[featureFlag.evaluation.reason]}</> : '--'}
+                        {featureFlag.active ? (
+                            <>{featureFlagMatchMapping[featureFlag.evaluation.reason as FeatureFlagMatchReason]}</>
```
Comment (dmarticus) [inline frontend/src/scenes/persons/RelatedFeatureFlags.tsx:128]: I typecast here to make the red squiggle go away in my editor.

Comment (havenbarnes) [reply frontend/src/scenes/persons/RelatedFeatureFlags.tsx:128]: If it doesn't cause other problems, why don't we just change the type here to `FeatureFlagMatchReason` instead? https://github.com/PostHog/posthog/blob/chore/filter-unrelated-flags-in-person-groups-ui/frontend/src/scenes/persons/relatedFeatureFlagsLogic.ts#L20

**Review by havenbarnes — APPROVED** [review]
Looks great!

File: frontend/src/scenes/persons/RelatedFeatureFlags.tsx (hunk 1 above)
Comment (dmarticus) [reply frontend/src/scenes/persons/RelatedFeatureFlags.tsx:128]: great call, I didn't look hard enough to realize that was the problem.

**Pull request merged**